librz/core/gdiff.c

```diff
@@ -137,18 +220,37 @@ RZ_API void rz_core_diff_show(RzCore *c, RzCore *c2) {
 		case RZ_ANALYSIS_FCN_TYPE_SYM:
 			switch (f->diff->type) {
 			case RZ_ANALYSIS_DIFF_TYPE_MATCH:
-				match = "MATCH";
-				break;
 			case RZ_ANALYSIS_DIFF_TYPE_UNMATCH:
-				match = "UNMATCH";
+				is_new = false;
 				break;
 			default:
-				match = "NEW";
-				f->diff->dist = 0;
+				is_new = true;
+			}
+			if (json) {
+				double dist = f->diff->dist;
+				pj_o(pj);
+				pj_kd(pj, "distance", f->diff->dist);
+				pj_ks(pj, "type", dist >= 1.0 ? "MATCH" : (dist >= 0.5 ? "SIMILAR" : (is_new ? "NEW" : "UNMATCH")));
+				if (f->name) {
+					pj_ko(pj, "original");
+					pj_ks(pj, "name", f->name);
+					pj_kn(pj, "addr", f->addr);
+					pj_kn(pj, "size", rz_analysis_function_linear_size(f));
```

Comment:
Are you sure the linear size makes sense here? When you have 100% identical function code and bbs but one has a basic block moved way further away, the linear size difference will be huge. The realsize in this case would give you the sum of the sizes of all basic blocks.

Reply:
But I also don't know what "size" the diff actually calculates.

Reply:
what is the function name then?

Reply:
`rz_analysis_function_realsize()`. But I don't know what `f->diff->size` is supposed to contain so I guess it should be consistent with that.

Reply:
https://github.com/rizinorg/rizin/blob/dev/librz/analysis/diff.c#L205
that function diffs 2 functions and checks their similarity.

Reply:
I think, since we later call `rz_analysis_diff_bb(analysis, fcn, fcn2);`, it makes sense to use the `realsize` here, as the size of all basic blocks.

Reply:
ok, then i need to change also the one in the diff code.